ngraph_bridge/ngraph_builder.h

```diff
@@ -144,12 +146,42 @@ class Builder {
                 const ngraph::element::Type>>&
   TF_NGRAPH_CONST_MAP();
 
+  template <class TOpType, class... TArg>
+  static std::shared_ptr<TOpType> ConstructNgNode(const std::string& op_name,
+                                                  TArg&&... Args) {
+    auto ng_node = std::make_shared<TOpType>(std::forward<TArg>(Args)...);
+    ng_node->set_friendly_name(op_name);
+    ng_node->add_provenance_tag(op_name);
+    InsertInLogs(op_name, ng_node->get_name());
+    return ng_node;
+  }
+
+  static void PrintTfNodeToNgNodeConversionTable() {
+    if (config::IsLoggingPlacement()) {
+      for (const auto& itr : tf_node_to_ng_node_) {
+        cout << "TF_to_NG: " << itr.first << " --> " << ng::join(itr.second)
+             << "\n";
+      }
+    }
+    tf_node_to_ng_node_.clear();
+  }
+
  private:
   static void ComputeScaleOffsetFolded(const uint& num_bits,
                                        const bool& unsigned_type,
                                        const bool& scaled, const int min_range,
                                        const int max_range, float* scale,
                                        int* offset);
+  static std::map<string, std::vector<string>> tf_node_to_ng_node_;
+  static std::mutex tf_to_ng_table_lock_;
```

Comment:
In case there are 2 instances running, `tf_node_to_ng_node_` being a global is under contention will be overwritten unsafely.
In test frameworks like TF's test_runner, tests might be running parallelly (not sure, need to check). Instances were observed where `tf_node_to_ng_node_` is corrupted. Hence locking it. Locking can be done in `InsertInLogs` and `PrintTfNodeToNgNodeConversionTable` only (instead of across the for loop).

Need to discuss this.